File: itunes.rb
require "open-uri"
require "json"

res = JSON.parse(open("https://itunes.apple.com/search?term=kinggnu/").map {|r| r}.join)
search_results = res["results"]
puts search_results.select {|r| r["trackName"] === "Hakujitsu"}.first["collectionViewUrl"]

File: itunes.php
<?php
require_once("vendor/autoload.php");

$ch = curl_init('https://itunes.apple.com/search?term=kinggnu/');
curl_setopt_array($ch, [CURLOPT_RETURNTRANSFER => true]);
$json           = json_decode(curl_exec($ch), true);
$search_results = $json["results"];

foreach($search_results as $v) {
  if ($v["trackName"] == "Hakujitsu") {
    echo($v["collectionViewUrl"]);
  }
}

